streams/jvm/src/main/scala/scalaz/zio/stream/platform.scala

```diff
@@ -0,0 +1,51 @@
+package scalaz.zio.stream
+
+import java.io.{ InputStream, OutputStream }
+
+import scalaz.zio._
+import scalaz.zio.blocking._
+
+trait ZStreamPlatformSpecific {
+
+  /**
+   * Uses the provided `TaskR` value to create a [[ZStream]] of byte chunks, backed by
+   * the resulting `InputStream`. When data from the `InputStream` is exhausted,
+   * the stream will close it.
+   */
+  def fromInputStream[R](
+    is: TaskR[R, InputStream],
+    chunkSize: Int = ZStreamChunk.DefaultChunkSize
+  ): ZStreamChunk[R with Blocking, Throwable, Byte] =
+    ZStreamChunk {
+      ZStream.managed(is.toManaged(is => effectBlocking(is.close()).orDie)) { is =>
+        effectBlocking {
+          val buf       = Array.ofDim[Byte](chunkSize)
+          val bytesRead = is.read(buf)
+
+          if (bytesRead < 0) None
```

Comment:
```suggestion
          if (bytesRead <= 0) None
```

Reply:
Hmm @regiskuckaertz I'm not sure that's accurate as the javadoc says that `-1` is the sentinel.